### PR TITLE
Optimize `COUNT(1)`: Change the sentinel value's type for COUNT(*) to Int64

### LINKED ISSUE #9943: `COUNT(1)` is different from `COUNT(*)`
### Describe the bug

As per the standard `COUNT(1)` is equivalent to the more common `COUNT(*)`, however this is not the case now, since the latter has the following sentinel value:
https://github.com/apache/arrow-datafusion/blob/4bd7c137e0e205140e273a7c25824c94b457c660/datafusion/expr/src/utils.rs#L43-L45
whereas the former resolves to `COUNT(Int64(1))`.

In turn this means this optimization is not being used
https://github.com/apache/arrow-datafusion/blob/4bd7c137e0e205140e273a7c25824c94b457c660/datafusion/core/src/physical_optimizer/aggregate_statistics.rs#L158-L162

### To Reproduce

```bash
gruuya@markogrujics-MacBook-Pro datafusion-cli % DATAFUSION_EXECUTION_COLLECT_STATISTICS=true datafusion-cli
DataFusion CLI v36.0.0
❯ create external table hits stored as parquet location 'benchmarks/data/hits.parquet';
0 rows in set. Query took 0.030 seconds.

❯ explain select count(*) from hits;
+---------------+---------------------------------------------------------------+
| plan_type     | plan                                                          |
+---------------+---------------------------------------------------------------+
| logical_plan  | Aggregate: groupBy=[[]], aggr=[[COUNT(UInt8(1)) AS COUNT(*)]] |
|               |   TableScan: hits projection=[]                               |
| physical_plan | ProjectionExec: expr=[99997497 as COUNT(*)]                   |
|               |   PlaceholderRowExec                                          |
|               |                                                               |
+---------------+---------------------------------------------------------------+
2 rows in set. Query took 0.039 seconds.

❯ explain select count(1) from hits;
+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| logical_plan  | Aggregate: groupBy=[[]], aggr=[[COUNT(Int64(1))]]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
|               |   TableScan: hits projection=[]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
| physical_plan | AggregateExec: mode=Final, gby=[], aggr=[COUNT(Int64(1))]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
|               |   CoalescePartitionsExec                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
|               |     AggregateExec: mode=Partial, gby=[], aggr=[COUNT(Int64(1))]                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
|               |       ParquetExec: file_groups={12 groups: [[Users/gruuya/Splitgraph/arrow-datafusion/benchmarks/data/hits.parquet:0..1231664704], [Users/gruuya/Splitgraph/arrow-datafusion/benchmarks/data/hits.parquet:1231664704..2463329408], [Users/gruuya/Splitgraph/arrow-datafusion/benchmarks/data/hits.parquet:2463329408..3694994112], [Users/gruuya/Splitgraph/arrow-datafusion/benchmarks/data/hits.parquet:3694994112..4926658816], [Users/gruuya/Splitgraph/arrow-datafusion/benchmarks/data/hits.parquet:4926658816..6158323520], ...]} |
|               |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
2 rows in set. Query took 0.011 seconds.
```

Given that `count(*)` basically just prints out the metadata, while `count(1)` does some additional work, there's also a an apparent performance implication, albeit negligible, with `count(1)` being 10x slower:
```
❯ select count(*) from hits;
+----------+
| COUNT(*) |
+----------+
| 99997497 |
+----------+
1 row in set. Query took 0.004 seconds.

❯ select count(1) from hits;
+-----------------+
| COUNT(Int64(1)) |
+-----------------+
| 99997497        |
+-----------------+
1 row in set. Query took 0.043 seconds.
```

### Expected behavior

`COUNT(1)` and `COUNT(*)` should be interchangable and result in the same logical and physical plans (save for the column alias).

### Additional context

I can get the expected behavior by explicit casting, though the column name is off then
```bash
❯ explain select count(1::tinyint unsigned) from hits;
+---------------+----------------------------------------------------------------------+
| plan_type     | plan                                                                 |
+---------------+----------------------------------------------------------------------+
| logical_plan  | Aggregate: groupBy=[[]], aggr=[[COUNT(UInt8(1)) AS COUNT(Int64(1))]] |
|               |   TableScan: hits projection=[]                                      |
| physical_plan | ProjectionExec: expr=[99997497 as COUNT(*)]                          |
|               |   PlaceholderRowExec                                                 |
|               |                                                                      |
+---------------+----------------------------------------------------------------------+
2 rows in set. Query took 0.007 seconds.

❯ select count(1::tinyint unsigned) from hits;
+----------+
| COUNT(*) |
+----------+
| 99997497 |
+----------+
1 row in set. Query took 0.003 seconds.
```

### PR DESCRIPTION
**Pull request by gruuya**
## Which issue does this PR close?

Closes #9943.

## Rationale for this change

Make `COUNT(1)` equivalent to `COUNT(*)`.

## What changes are included in this PR?

Change the type of `COUNT_STAR_EXPANSION` to `ScalarValue::Int64`.

## Are these changes tested?

Yes, with revised existing tests and a new one that tests the optimization for `count(1)` case.

### CODE REVIEW

**Comment by gruuya** [comment]
Might as well try whether /benchmark works as intended now.

File: datafusion/expr/src/utils.rs [hunk 1]
```diff
@@ -42,7 +42,7 @@ use sqlparser::ast::{ExceptSelectItem, ExcludeSelectItem, WildcardAdditionalOpti
 
 ///  The value to which `COUNT(*)` is expanded to in
 ///  `COUNT(<constant>)` expressions
-pub const COUNT_STAR_EXPANSION: ScalarValue = ScalarValue::UInt8(Some(1));
+pub const COUNT_STAR_EXPANSION: ScalarValue = ScalarValue::Int64(Some(1));
```
Comment (gruuya) [inline datafusion/expr/src/utils.rs:45]: This does however result in `count(1)` being auto-named to `count(*)` now.
```bash
❯ explain select count(1) from hits;
+---------------+---------------------------------------------------+
| plan_type     | plan                                              |
+---------------+---------------------------------------------------+
| logical_plan  | Aggregate: groupBy=[[]], aggr=[[COUNT(Int64(1))]] |
|               |   TableScan: hits projection=[]                   |
| physical_plan | ProjectionExec: expr=[99997497 as COUNT(*)]       |
|               |   PlaceholderRowExec                              |
|               |                                                   |
+---------------+---------------------------------------------------+
2 row(s) fetched.
Elapsed 0.263 seconds.

❯ select count(1) from hits;
+----------+
| COUNT(*) |
+----------+
| 99997497 |
+----------+
1 row(s) fetched.
Elapsed 0.015 seconds.
```
On the other hand that might be fine as it reduces ambiguity.

Comment (alamb) [reply datafusion/expr/src/utils.rs:45]: > This does however result in count(1) being auto-named to count(*) now.

FWIW, this is what happens on main with `count(uint8)` (the same thing)

```sql

❯ select count(arrow_cast(1, 'UInt8')) from (values (1));
+----------+
| COUNT(*) |
+----------+
| 1        |
+----------+
1 row(s) fetched.
Elapsed 0.002 seconds.
```

Though admittedly almost no one would actually type `count(arrow_cast(1, 'UInt8'))` so it is likely not a big deal.

> That might be fine as it reduces ambiguity.

I agree having the `COUNT(*)` in the plan actually helps as it makes it clearer when the fast path is being used. 

Let's start with this and if someone else has a different opinion we can make another PR

**Review by alamb — APPROVED** [review]
Thanks @gruuya  -- this PR makes senes to me. I have definitely seen people do `SELECT COUNT(1)` rather than `SELECT COUNT(*)` so having them have the same performance seems good 👨‍🍳 👌

**Review by Jefffrey — APPROVED** [review]
Makes sense to me, nice find :+1:

**Comment by alamb** [comment]
Thanks again @gruuya

**Pull request merged**